pvlib/pvsystem.py

```diff
@@ -1466,6 +1439,90 @@ def get_iam(self, aoi, iam_model='physical'):
+        wind_speed : numeric
+            Wind speed [m/s]
+
+        model : str
+            Supported models include ``'sapm'``, ``'pvsyst'``,
+            ``'faiman'``, ``'fuentes'``, and ``'noct_sam'``
+
+        **kwargs
+            Extra arguments passed to the model function.
+
+        Returns
+        -------
+        numeric
+            values in degrees C.
+
+        Notes
+        -----
+        Some temperature models have requirements for the input types;
+        see the documentation of the underlying model function for details.
+        """
+        def _build_tcell_args(keys):
+            # convenience wrapper to avoid passing args 2 and 3 every call
+            return _build_args(keys, self.temperature_model_parameters,
+                               'temperature_model_parameters')
+        if model == 'sapm':
+            func = temperature.sapm_cell
+            required = _build_tcell_args(['a', 'b', 'deltaT'])
+            optional = _build_kwargs(['irrad_ref'],
+                                     self.temperature_model_parameters)
+        elif model == 'pvsyst':
+            func = temperature.pvsyst_cell
+            required = tuple()
+            optional = {
+                # TODO remove 'eta_m' after deprecation of this parameter
+                **_build_kwargs(['eta_m', 'module_efficiency',
+                                 'alpha_absorption'],
+                                self.module_parameters),
+                **_build_kwargs(['u_c', 'u_v'],
+                                self.temperature_model_parameters)
+            }
+        elif model == 'faiman':
+            func = temperature.faiman
+            required = tuple()
+            optional = _build_kwargs(['u0', 'u1'],
+                                     self.temperature_model_parameters)
+        elif model == 'fuentes':
+            func = temperature.fuentes
+            required = _build_tcell_args(['noct_installed'])
+            optional = _build_kwargs([
+                'module_height', 'wind_height', 'emissivity', 'absorption',
+                'surface_tilt', 'module_width', 'module_length'],
+                self.temperature_model_parameters)
+            # default to using the Array attribute, but allow user to override
+            # with a custom surface_tilt value in temperature_model_parameters
+            if 'surface_tilt' not in optional:
+                optional['surface_tilt'] = self.surface_tilt
+        elif model == 'noct_sam':
+            func = temperature.noct_sam
+            required = _build_tcell_args(['noct', 'module_efficiency'])
+            optional = _build_kwargs(['transmittance_absorptance',
+                                      'array_height', 'mount_standoff'],
+                                     self.temperature_model_parameters)
+        else:
+            raise ValueError(f'{model} is not a valid cell temperature model')
+
+        # allow kwargs to override
+        optional.update(kwargs)
```

Comment:
Is there a compelling use case for kwargs here?

kwargs may be a necessary evil in `PVSystem.get_cell_temperature` due to `effective_irradiance`, but I'm -1 on allowing kwargs to arbitrarily update the parameters here. The object layer is already complicated and difficult to maintain. I also keep hitting bugs in my own code due to kwargs.

Reply:
I also don't see that this is tested.

Reply:
Not sure why I thought this was a good idea.  Maybe I forgot that `temperature_model_parameters` existed.  I agree that kwargs is unnecessary and undesirable here, and it's now removed.